packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -307,33 +309,16 @@ module.exports = (
         response: {
           schema: isA.object({
             exists: isA.boolean().required(),
+            hint: isA.string().optional().allow(null),
             estimatedSyncDeviceCount: isA.number().optional(),
           }),
         },
       },
       async handler(request) {
         log.begin('recoveryKeyExists', request);
 
-        const { email } = request.payload;
-
-        let uid;
-        if (request.auth.credentials) {
-          uid = request.auth.credentials.uid;
-        }
-
-        if (!uid) {
-          // If not using a sessionToken, an email is required to check
-          // for an account recovery key. This occurs when checking from the
-          // password reset page and allows us to redirect the user to either
-          // the regular password reset or account recovery password reset.
-          if (!email) {
-            throw errors.missingRequestParameter('email');
-          }
-
-          await customs.check(request, email, 'recoveryKeyExists');
-          ({ uid } = await db.accountRecord(email));
-        }
-        const exists = await db.recoveryKeyExists(uid);
+        const uid = request.auth.credentials.uid;
+        const recoveryKeyRecordWithHint = await db.getRecoveryKeyHint(uid);
```

Comment:
nit: The variable name is pretty clear, but database function name is a little confusing. Maybe they could be named similar.